### ISSUE TITLE
Every wallet I try to recover to has corrupted databases...

### ISSUE THREAD

**Issue by bakd247** [bitcoin-dot-org/bitcoin.org#1379]
I have tried every single bitcoin wallet I can find to try and sync my backup file and every time an internal fatal error occurs half way through syncing....I am convinced that my bitcoin is gone for good and that bitcoin doesnt work...mainly because every person I talk to has no idea what i can do either....

**Comment by bakd247** [comment]
$75 worth of bitcoin invested and not a single wallet will let me spend it because nothing syncs to the chain...I am so pissed off right now that I have decided to just secure my money in gold instead of a coin that constantly crashes...

**Comment by luke-jr** [comment]
@bakd247 If nothing syncs, then your PC is probably defective and should be replaced. Be sure to backup your wallet in the meantime.

**Comment by bakd247** [comment]
Theres nothing wrong with my PC..its a brand new and gaming PC and everything works and syncs fine except for bitcoin....

**Comment by bakd247** [comment]
I've been trading litecoin and cannabis coin all weekend with no problems at all

**Comment by bakd247** [comment]
I've already ran a memory check, ssd check, cpu, gpu and motherboard check....i built this computer after years of experience building pcs....theres nothing wrong with it....only problem I am having is a bitcoin wallet software issue....my backup isn't working on any wallet

**Comment by laanwj** [comment]
If everything fails on you, the only common factor is you.

**Comment by bakd247** [comment]
Ok ...i didn't write the software for bitcoin core....only problem here is bitcoin core....

Sent from my Windows Phone
________________________________

**Comment by MarcoFalke** [comment]
@bakd247 Usually this failure in your debug.log indicates a hardware failure. If you are certain that your hardware is functioning properly, this is worth to take a look into...

Can you reproduce when you create a new user profile in Windows? Or when trying this in a VM?

Would you mind to upload the full debug.log?

**Comment by bakd247** [comment]
I deleted everything bitcoin except my saved backup dat file and am trying xt again....version 13 told me that plus plus asked the software to close in an unusual way and said to contact the software developers about it...then it shut down....stuck at 1 year 45 weeks left of sync and my transactions didn't start till June of this year....so its the wallets not working in windows....every scan if everything comes out fine....bitcoin is unusable...no matter what I do....im about to just call it a loss and be done all together with bitcoin

**Comment by bakd247** [comment]
Now in xt...only the coins I have earned through minning are pending....every transaction I exchanged USD for through coin base and circle is "conflicted"

**Comment by bakd247** [comment]
Already contacted coinbase and circle about it with no response

**Comment by bakd247** [comment]
F#*% me...i got it....note never upgrade after a crash....all coins pending in version 12.0.   version 13 xt and 12.1 will not load a previous version dat back up file

**Comment by bakd247** [comment]
Ok so how do I unlock my wallet...im getting a run time error and I would think I would need to enter a password or something...wont finish syncing the blockchain

**Comment by bakd247** [comment]
Debug file says level a read failure: Corruption: Block checksum mismatch
Corruption:block checksum mismatch
Error reading from database: Database corrupted

**Comment by bakd247** [comment]
And preallocating up to position 0x00000in rev00192.dat keeps repeating for a while towards the end before it said database corrupted

**Comment by bakd247** [comment]
Stuck at block 329392

**Comment by luke-jr** [comment]
This has nothing to do with the bitcoin.org website. Also, XT is known to be broken. Don't use it.

**Comment by bakd247** [comment]
Alll bitcoin core is broken...

Sent from my Windows Phone
________________________________

**Comment by bakd247** [comment]
Version 12 is the only one that will put my coin as pending when I recover....and it crashes at the same block everytime with over 1 year 48 weeks left...debug says database is corrupted....and its doing it on 3 different pcs....no matter what I do, my coin is stuck...i am convinced I lost my bitcoin

**Comment by bakd247** [comment]
Most screwed up part about all this is the coin I paid USD for is conflicted with any other wallet...only version 12 puts my paid for coin as pending....I'm also convinced that coinbase generated a transaction and my coin was never any good to begin with...until I can get rid of and be done with bitcoin, nothing will change my mind about that...i dont care about what everyone else says...it only takes one bad experience for me to not trust something or someone...and I am convinced my bitcoin is gone because of a hardware issue...backups are pointless if the chain wont sync

**Comment by bakd247** [comment]
Trying everything again on new partition as administrator....its the only thing left....I think the problem is that version 12 doesn't work with windows 10 architecture....because everyone who has ever been able to recover a wallet that I have seen online is using Windows 8 or 7...im betting my $42 worth of bitcoin that's still hung up that I am right

**Comment by bakd247** [comment]
Yep...bitcoin ate $42 worth of coin...not a single wallet in existence that will sync with my backups....done even thinking about bitcoin.....worse experience ever

**Comment by bakd247** [comment]
I knew I wasn't crazy....bitcoin wont connect because of a c plus plus error...has nothing to do with hardware over heating or anything anyone has suggested....bitcoin was basically stopped by windows defender and I had to open the port manually

**Comment by bakd247** [comment]
And roll back to visual studio 2013...no wallets work with visual studio 2015

**Comment by bakd247** [comment]
Turns out all wallets have to have visual studios c plus plus 2010 in order to work...any later version makes the wallet crash....

**Comment by gruvin** [comment]
I've been running a Bitcoin.org "full node" for years now. Today, the database got corrupted for the umpteenth time. (Power failure and unfortunate UPS fault, despite recent battery replacement and other such routine maintenance and testing. :-/ ) I have to run it on dedicated (bitcoin only) hardware, because the chances of some other software or OS kernel fault resulting in forced resets and thus almost ALWAYS Bitcoin.org database corruption are too high a risk. This is IMO RIDICULOUS, even shameful.

For those reasons -- and only those reasons, in my opinion -- the software frankly just SUX. It's clearly not even remotely fault tolerant, from a data security point of view -- besides what come from the overall hive. The slightest underlying OS fault (or internal Bitcoin.org bug) can and often DOES toast its database -- all several hundred gigabytes of it, essentially. So then you're back to WEEKS (and soon months, no doubt) of waiting for the thing to sync from scratch, before you can can even use it, let alone access your wallet.  

They say it's our responsibility to maintain offline backups, which I do. But even those cannot be performed from a live, running system. No attempts for "shadow copy" index management exist, to my knowledge. Thus, backups can only proceed with the software completely offline. Full backups take ages to complete, for obvious reasons ie. blockchain size. I do weekly Rsync and bi-monthly full backups -- because Rsync will of course replace good data with bad, if I don't realise data corruption before doing the sync to offline storage. I have had to spend money on multiple dedicated backup drives and stick to rigorous backup schedules, just to keep the data stable of the medium to long term.

It bows my mind that hundreds of millions of transactions are handled by software that has made absolutely no attempt to create secure, fault tolerant database storage -- even when multiple, open source and free technologies for same have been available for YEARS. MySQL, SQLLite (potentially) PostgreSQL, etc to name a few. Berkley DB? Are you freakin' serious? OMG! 

The Bitcoin.org programmers keep adding this feature and that, without first fixing the primary foundation of the software -- the data storage layer -- you know, that part of the software responsible for storing and maintaining the blockchain itself. Isn't that the entire POINT of the thing? W T F people! :-/ This could be fixed in a weekend. (Wish I knew how. I would do it myself.)

It long ago stopped making sense for me to run a full node, besides a rapidly fading keen desire to support Bitcoin in general. It sure as hell stopped making sense to store my coin in a Bitcoin.org node's wallet! 

Fewer full nodes exist now than ever. Gee, I wonder why? (not) It's beyond belief.

[Andreas M. Antonopoulos](https://antonopoulos.com)  says that Bitcoin is, "failing gracefully". Well, not really. There is nothing, "graceful" in my mind about willingingly and blindly basing an entire economy on pathetically and needlessly inappropriate database layer. Again, WHAT THE #@$!! Nope. That's not graceful. That's a cluster fuck.

**In summary, Bitcoin.org "full node" software is unreliable and thus IMO simply CRAP. I recommend not using it and absolutely NOT trusting it with your wallet. Use something else.**
